hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterRpc.java

```diff
@@ -1901,4 +1904,27 @@ private DFSClient getFileDFSClient(final String path) {
     }
     return null;
   }
+
+  @Test
+  public void testMkdirsWithCallerContext() throws IOException {
+    GenericTestUtils.LogCapturer auditlog =
+        GenericTestUtils.LogCapturer.captureLogs(FSNamesystem.auditLog);
+
+    // Current callerContext is null
+    assertNull(CallerContext.getCurrent());
+
+    // Set client context
+    CallerContext.setCurrent(
+        new CallerContext.Builder("clientContext").build());
+
+    // Create a directory via the router
+    String dirPath = "/test_dir_with_callercontext";
+    FsPermission permission = new FsPermission("755");
+    routerProtocol.mkdirs(dirPath, permission, false);
+
+    // The audit log should contains "callerContext=clientContext,clientIp:"
+    assertTrue(auditlog.getOutput()
+        .contains("callerContext=clientContext,clientIp:"));
```

Comment:
Anyway we can check for the actual IP?

Reply:
Had thought  about this. DFSClient & Client do not expose ip, and TestAuditLogger & TestAuditLogs do not check client ip. So do you have any suggestions?

Reply:
Or just keep it that way, and do not modify UT

Reply:
The only issue I see is to grab random logs but I guess is fine.

Reply:
Sorry, do not understand it.
I think If we check the actual IP, we should get the client actual IP, e.g "w.x.y.z", and then check the audit log, it should contain "callerContext=clientContext,clientIp:w.x.y.z", is it right?
Now it's hard to get client ip.

Reply:
Correct, grabbing the proper Client IP is not trivial and error prone so I'm fine with this.